packages/web/docs/src/pages/docs/gateway/other-features/plugins.mdx

```diff
@@ -0,0 +1,618 @@
+added automatically based on your gateway configuration.
+
+## Configuration
+
+You can add plugins to your gateway using the `plugins` option.
+
+It have to be a function, which will be called each time the gateway have update its configuration.
+For example, if polling is enabled, this function will be called for each poll.
+
+Most Hive Gateway plugins takes an object as a parameter, and expect some common components like a
+`logger`, a `pubsub`, etc... Those components are given in parameters to the `plugins` function. It
+is advised to spread the plugin's factory context into the plugins options, this way plugins will
+have access to all components they need.
+
+```ts filename="gateway.config.ts"
+import { defineConfig } from '@graphql-hive/gateway'
+import useNewRelic from '@graphql-mesh/plugin-newrelic'
+import { useSOFA } from '@graphql-yoga/plugin-sofa'
+
+export const gatewayConfig = defineConfig({
+  plugins: ctx => [
+    useSOFA({
+      ...ctx,
+      basePath: '/rest',
+      swaggerUIEndpoint: '/rest/docs'
+    }),
+    useNewRelic({
+      ...ctx
+    })
+  ]
+})
+```
+
+## Custom Plugins
+
+You can take advantage of the plugin system to create your own custom plugin. This allows you to
+integrate your gateway with any technology that we don't officially support, being public or private
+to your company.
+
+A good entry-point for discovering how to write Gateway plugins is to look at the source code of the
+existing plugins maintained by us.
+
+A plugin is an object that can define methods to hook into different phases or events of the gateway
+lifecycle. Each hook will receive a single parameter, the hook payload, containing data related to
+the hooked phase and functions allowing you to modify the behavior of this phase.
+
+There are 2 categories of hooks:
+
+- Gateway lifecycle hooks: they are related to the lifecycle or events of the server itself and is
+  not related to a specific request.
+- Request lifecycle hooks: they are tight to a specific phase of a specific request. Those hooks
+  always give access to the related request in there payload.
+
+### Gateway lifecycle
+
+Gateway lifecycle allows to setup long running services, initialize in-memory cache or stores, react
+to general events.
+
+<div className="[&_a]:text-primary [&_a]:underline [&_a:hover]:no-underline [&_a]:decoration-from-font [&_a]:[text-underline-position:from-font]">
+
+```mermaid
+stateDiagram-v2
+  onYogaInit: <a href="#onyogainit">onYogaInit</a>
+  onPluginInit: <a href="#onplugininit">onPluginInit</a>
+  onDispose: <a href="#ondispose">onDispose</a>
+  listening: Listening for HTTP requests
+
+  [*] --> onYogaInit
+  onYogaInit --> onPluginInit
+  onPluginInit --> listening
+  listening --> onDispose: Gateway is shutting down
+
+  note left of listening: <a href="#onschemachange">onSchemaChange</a> called for each new schema loaded
+  note left of listening: <a href="#onfetch">onFetch</a> called for each outgoing HTTP request
+```
```

Comment:
Not sure about how much this diagram is clear.

@Urigo what do you think ?

I'm not sure how to represent the fact that both `onFetch` and `onSchemaChange` can be called at any time during the lifetime of the gateway.